predict.py

```diff
@@ -4,33 +4,48 @@
 import torch
 from cog import BasePredictor, Input, Path
 from diffusers import (
-    StableDiffusionPipeline,
-    PNDMScheduler,
-    LMSDiscreteScheduler,
     DDIMScheduler,
-    EulerDiscreteScheduler,
-    EulerAncestralDiscreteScheduler,
     DPMSolverMultistepScheduler,
+    EulerAncestralDiscreteScheduler,
+    EulerDiscreteScheduler,
+    LMSDiscreteScheduler,
+    PNDMScheduler,
+    StableDiffusionPipeline,
 )
 from diffusers.pipelines.stable_diffusion.safety_checker import (
     StableDiffusionSafetyChecker,
 )
 
+from weights_downloader import WeightsDownloader
+
 # MODEL_ID refers to a diffusers-compatible model on HuggingFace
 # e.g. prompthero/openjourney-v2, wavymulder/Analog-Diffusion, etc
-MODEL_ID = "stabilityai/stable-diffusion-2-1"
 MODEL_CACHE = "diffusers-cache"
+
+SD_MODEL_CACHE = os.path.join(MODEL_CACHE, "models--stabilityai--stable-diffusion-2-1")
+MODEL_ID = "stabilityai/stable-diffusion-2-1"
+SD_URL = "https://weights.replicate.delivery/default/stable-diffusion/stable-diffusion-2-1.tar"
+
+SAFETY_CACHE = os.path.join(
+    MODEL_CACHE, "models--CompVis--stable-diffusion-safety-checker"
+)
 SAFETY_MODEL_ID = "CompVis/stable-diffusion-safety-checker"
+SAFETY_URL = "https://weights.replicate.delivery/default/stable-diffusion/stable-diffusion-safety-checker.tar"
+
 
 class Predictor(BasePredictor):
     def setup(self):
         """Load the model into memory to make running multiple predictions efficient"""
+
         print("Loading pipeline...")
+        WeightsDownloader.download_if_not_exists(SAFETY_URL, SAFETY_CACHE)
         safety_checker = StableDiffusionSafetyChecker.from_pretrained(
```

Comment:
This should probably include a revision, same as https://github.com/replicate/cog-deepfloyd-if/pull/2